common/stack_tracer.go

```diff
@@ -60,6 +62,94 @@ func TraceAsmCaller(
 	return src, nil
 }
 
+// TraceAllAsmCaller correctly tracks all possible function calls in the execution stack.
+func TraceAllAsmCaller(
+	filePath string,
+	graph asmparser.CallGraph,
+	function string,
+	endCond func(string) bool,
+) ([]*analyzer.CallStack, error) {
+	var segment asmparser.Segment
+	for _, seg := range graph.Segments() {
+		if seg.Label() == function {
+			segment = seg
+			break
+		}
+	}
+	if segment == nil {
+		return nil, fmt.Errorf("could not find %s in %s", function, filePath)
+	}
+	sources := make([]*lifo.Stack[asmparser.Segment], 0)
+	currentStack := lifo.Stack[asmparser.Segment]{}
+	seen := make(map[asmparser.Segment]bool)
+
+	var visit func(segment asmparser.Segment)
+
+	visit = func(segment asmparser.Segment) {
+		if seen[segment] {
+			return
+		}
+		seen[segment] = true
+		currentStack.Push(segment)
+
+		if endCond(segment.Label()) {
+			sources = append(sources, currentStack.Copy())
+		} else {
+			parents := graph.ParentsOf(segment)
+			// sort the parents for consistent output
+			slices.SortFunc(parents, func(a, b asmparser.Segment) int {
+				if a.Address() > b.Address() {
+					return 1
+				} else if a.Address() < b.Address() {
+					return -1
+				}
+				return 0
+			})
+
+			for _, seg := range parents {
+				visit(seg)
+			}
+		}
+
+		currentStack.Pop()
+		// We don't want to revisit the runtime and internal packages as the number of paths can be up to billions.
+		if !strings.Contains(segment.Label(), "runtime") &&
+			strings.Contains(segment.Label(), "internal") {
```

Comment:
I think you forgot to add `!` before `strings.Contains(segment.Label(), "internal")`